amplfi/train/models/flow.py

```diff
@@ -124,6 +126,66 @@ def on_test_epoch_start(self):
         self.test_results: list[Result] = []
         self.idx = 0
 
+    def _get_log_prior_dict(self, trainer=None):
+        """
+        Get the log prior for each parameter in the model.
+        """
+        if trainer is not None:
+            self.trainer = trainer
+        log_prior_dict = {}
+        sf = self.trainer.datamodule.waveform_sampler.parameter_sampler # need this to get the prior for some parameters
+
+        for param in self.inference_params:
+            if param in sf.parameters:
+                # some priors live in the ```waveform_sampler.parameter_sampler``` object
+                # {chirp_mass': Uniform(low: 10.0, high: 100.0),
+                # 'mass_ratio': Uniform(low: 0.125, high: 0.9990000128746033),
+                # 'distance': PowerLaw(),
+                # 'inclination': Sine(),
+                # 'phic': Uniform(low: 0.0, high: 6.2831854820251465),
+                # 'chi1': DeltaFunction(),
+                # 'chi2': DeltaFunction()}
+                log_prior_dict[param] = sf.parameters[param]
+            else:
+                # others are seperately
+                # waveform_sampler."name"
+                # 'dec': Cosine(),
+                # 'psi': Uniform(low: 0.0, high: 3.140000104904175),
+                # 'phi': Uniform(low: -3.140000104904175, high: 3.140000104904175)
+                log_prior_dict[param] = getattr(self.trainer.datamodule.waveform_sampler, param)
+        # add low and high boundaries (for some parameters these values are missing)
+        log_prior_dict['distance'].low = 100
```

Comment:
@asasli Im wondering if, instead of using the prior ranges, we can simply call `prior.prob(sample)`, and discard the elements that have 0 probability. That way, this functionality is robust to changes in the prior ranges, and we don't have to add these prior specific `low` and `high` attributes. Do you think that would work?

Alternatively, we could add `low` and `high` attributes to each of the prior classes.